tests/rptest/tests/consumer_group_test.py

```diff
@@ -639,3 +646,146 @@ class OffsetFetchRequest_v5(Request):
+                    'enable.auto.offset.store': False,
+                })
+                self.consumer.subscribe([self.topic])
+                self.consumer.poll(0.5)
+                self.restart.clear()
+                self.restarted.set()
+
+            try:
+                msg = self.consumer.poll(timeout=1.0)
+                if msg is None:
+                    continue
+                if msg.error():
+                    self.logger.error(
+                        f"consumer {self.id} error - {msg.error()}")
+                    continue
+
+                self.consumer.store_offsets(msg)
+                with self.lock:
+                    self.last_consumed = msg.offset()
+
+            except Exception as e:
+                self.logger.error(f"consumer {self.id} error - {e}")
+
+        self.consumer.close()
+
+    def get_last_consumed(self):
+        with self.lock:
+            return self.last_consumed
+
+    def restart_consumer(self):
+        self.restart.set()
+        self.restarted.wait()
+        self.restarted.clear()
+
+
+class ConsumerGroupStaticMembersRebalance(RedpandaTest):
+    @cluster(num_nodes=4)
+    def test_rejoining(self):
+        topic = TopicSpec(name="test-topic-1", partition_count=20)
+        DefaultClient(self.redpanda).create_topic(topic)
+        group_id = "test-group-1"
+
+        producer = KgoVerifierProducer(self.test_context, self.redpanda,
+                                       topic.name, 128, 5000000)
+        producer.start()
+        consumers = []
+
+        for c_id in range(20):
+            consumers.append(
+                TestConsumer(bootstrap_servers=self.redpanda.brokers(),
+                             group=group_id,
+                             topic=topic.name,
+                             id=c_id,
+                             logger=self.logger))
+
+        admin_client = AdminClient(
+            {"bootstrap.servers": self.redpanda.brokers()})
+
+        def get_group_description():
+            description = admin_client.describe_consumer_groups(
+                group_ids=[group_id])[group_id].result()
+            return description
+
+        def consumers_made_progress():
+            return all(c.get_last_consumed() is not None for c in consumers)
+
+        wait_until(consumers_made_progress, 60, 1)
+        before_restart = get_group_description()
+        self.logger.info("group state before restart: %s",
+                         before_restart.state)
+        assert before_restart.state == ConsumerGroupState.STABLE
+        assert len(before_restart.members) == 20
+        consumer_to_restart = random.choice(consumers)
+        last_consumed = consumer_to_restart.get_last_consumed()
+        consumer_to_restart.restart_consumer()
+        after_restart = get_group_description()
+        self.logger.info("group state after restart: %s", after_restart.state)
+        assert after_restart.state == ConsumerGroupState.STABLE
```

Comment:
So what is the difference between stop and restart? Can we in theory get a rebalance blip here if the consumer restarts slowly enough?

Reply:
yes, in theory yes, the difference between the static and dynamic membership is that in the case of static member consumer leaving group does not cause a rebalance.

Reply:
How long does it take to rebalance the consumer group in this scenario? I'm wondering how reliably this test would fail if there were a rebalance here. My guess is that the rebalance takes long enough that we would be able to observe the rebalance here with high enough probability.